src/tablib/formats/_ods.py

```diff
@@ -53,45 +54,18 @@ def dset_sheet(cls, dataset, ws):
             _offset = i
             _package.insert((sep[0] + _offset), (sep[1],))
 
-        for i, row in enumerate(_package):
-            row_number = i + 1
-            odf_row = table.TableRow(stylename=bold, defaultcellstylename='bold')
+        for row_number, row in enumerate(_package, start=1):
+            is_header = row_number == 1 and dataset.headers
+            style = bold if is_header else None
+            odf_row = table.TableRow(stylename=style)
+            ws.addElement(odf_row)
             for j, col in enumerate(row):
-                try:
-                    col = str(col, errors='ignore')
-                except TypeError:
-                    # col is already str
-                    pass
-                ws.addElement(table.TableColumn())
-
-                # bold headers
-                if (row_number == 1) and dataset.headers:
-                    odf_row.setAttribute('stylename', bold)
-                    ws.addElement(odf_row)
-                    cell = table.TableCell()
-                    p = text.P()
-                    p.addElement(text.Span(text=col, stylename=bold))
-                    cell.addElement(p)
-                    odf_row.addElement(cell)
-
-                # wrap the rest
+                if isinstance(col, numbers.Number):
+                    tc = table.TableCell(valuetype="float", value=col)
```

Comment:
Can we keep this named more descriptively as `cell` like elsewhere?

```suggestion
                    cell = table.TableCell(valuetype="float", value=col)
```